devito/passes/iet/definitions.py

```diff
@@ -470,30 +470,22 @@ def _map_function_on_high_bw_mem(self, site, obj, storage, devicerm, read_only=F
             if is_gpu_create(obj, self.gpu_create):
                 mmap = self.lang._map_alloc(obj)
 
-                cdims = make_zero_init(obj)
-                eq = Eq(obj[cdims], 0)
-
-                irs, _ = self.rcompile(eq)
-
-                init = irs.iet.body.body[0]
-
-                name = self.sregistry.make_name(prefix='init')
-                efunc = make_callable(name, init)
-                init = Call(name, efunc.parameters)
+                efuncs, init = make_zero_init(obj, self.rcompile, self.sregistry)
```

Comment:
Can we get read of `gpu-create` and just do `if is_transient` and have the zero init in the core no matter?

Reply:
as discussed offline, we would break user code

Reply:
We wouldn't we just need `is_gou_create` to return `true` for any `is_tansient` for now so that any transient function is directly allocated in gen code and add a deprecation warning

Reply:
if the users who have been using gpu-create don't start creating `is_transient` Functions, how can it possibly work?

I agree though that gpu-create should get deprecated

Reply:
Well yes why I said as a first step `is_gpu_create` should behave as now, but returns `true` as well for `is_transient` Function and throw a warning to tell people "stop using gpu-create and set your function with is_transient"